### PR TITLE
Tidy bound_identifiers library etc.

### PR DESCRIPTION
**Pull request by mshinwell**
The `bound_identifiers` library in Flambda 2 contained a lot of old cruft, which this patch cleans up.  There are also some associated cleanups in the `nominal` library.

### CODE REVIEW

File: middle_end/flambda2/bound_identifiers/bound_static.mli [hunk 1]
```diff
@@ -35,29 +39,26 @@ type t
 
 val empty : t
 
+(** Within a single value of type [t], symbols and code IDs bound by the
+    individual [Pattern]s are bound recursively, across all of the [Pattern]s.
+    However [Block_like] bindings are not bound recursively and are ordered in
+    the usual manner. (See [Simplify_static_const].) *)
```
Comment (lthls) [inline middle_end/flambda2/bound_identifiers/bound_static.mli:45]: I believe that the property we want is that all recursive cycles go through at least a code ID. So we could check that we're not creating bound patterns with more than one element, unless one of these elements is a `Code` pattern.
I'm not completely sure that we can enforce this invariant on the result of `From_lambda` without going through the code for the classic mode though.

**Comment by mshinwell** [comment]
@lthls is going to review this

File: middle_end/flambda2/terms/flambda.ml [hunk 2]
```diff
@@ -495,40 +494,40 @@ let rec named_must_be_static_consts (named : named) =
...
       ~block_like
   | Static_const _, Code _
   | (Code _ | Deleted_code), (Set_of_closures _ | Block_like _) ->
     Misc.fatal_errorf "Mismatch on variety of [Static_const]:@ %a@ =@ %a"
-      Bound_symbols.Pattern.print pat print_static_const_or_code
+      Bound_static.Pattern.print pat print_static_const_or_code
       static_const_or_code
 
-and match_against_bound_symbols_static_const_group :
+and match_against_bound_static__static_const_group :
```
Comment (lthls) [inline middle_end/flambda2/terms/flambda.ml:527]: Unintended double underscore ?

**Review by lthls — COMMENTED** [review]
Mostly good.

**Review by lthls — APPROVED** [review]

**Pull request merged**